packages/token/src/erc20/extensions/erc20_wrapper.cairo

```diff
@@ -0,0 +1,137 @@
+            let ok = token.transfer_from(caller, this, amount);
+            assert(ok, Errors::TRANSFER_FAILED);
+
+            let mut erc20_component = get_dep_component_mut!(ref self, ERC20);
+            erc20_component.mint(receiver, amount);
+            true
+        }
+
+        /// Burns wrapped tokens from caller and sends underlying tokens to `receiver`.
+        fn withdraw_to(
+            ref self: ComponentState<TContractState>, receiver: ContractAddress, amount: u256,
+        ) -> bool {
+            let caller = starknet::get_caller_address();
+            let this = starknet::get_contract_address();
+            assert(receiver != this, Errors::INVALID_RECEIVER);
+
+            let mut erc20_component = get_dep_component_mut!(ref self, ERC20);
+            erc20_component.burn(caller, amount);
+
+            let underlying = self.underlying();
+            let token = IERC20Dispatcher { contract_address: underlying };
+            let ok = token.transfer(receiver, amount);
+            assert(ok, Errors::TRANSFER_FAILED);
+            true
+        }
+    }
+
+    //
+    // Internal
+    //
+
+    #[generate_trait]
+    pub impl InternalImpl<
+        TContractState,
+        +HasComponent<TContractState>,
+        impl ERC20: ERC20Component::HasComponent<TContractState>,
+        +ERC20Component::ERC20HooksTrait<TContractState>,
+        +Drop<TContractState>,
+    > of InternalTrait<TContractState> {
+        /// Sets the underlying token address.
+        fn initializer(ref self: ComponentState<TContractState>, underlying: ContractAddress) {
+            let this = starknet::get_contract_address();
+            assert(underlying.is_non_zero(), Errors::INVALID_UNDERLYING_ADDRESS);
+            assert(underlying != this, Errors::INVALID_UNDERLYING_ADDRESS);
+            self.ERC20Wrapper_underlying.write(underlying);
+        }
+
+        /// Mints wrapped tokens to cover any underlying tokens that were transferred to this
+        /// contract by mistake, or acquired via rebasing mechanisms. Internal function that can be
+        /// exposed with access control if desired.
+        fn recover(ref self: ComponentState<TContractState>, account: ContractAddress) -> u256 {
+            let underlying = self.underlying();
+            let this = starknet::get_contract_address();
+
+            // get underlying balance held by this contract
+            let token = IERC20Dispatcher { contract_address: underlying };
+            let underlying_balance = token.balance_of(this);
+
+            // get current total supply of wrapped tokens
+            let mut erc20_component = get_dep_component_mut!(ref self, ERC20);
+            let total_supply = erc20_component.total_supply();
+
+            // calculate recoverable amount
+            let value = underlying_balance - total_supply;
+
+            // mint recoverable amount to specified account
+            erc20_component.mint(account, value);
```

Comment:
Why not inside `if (value > 0)` ? What's the point of minting 0 tokens?

Reply:
Reverting should be ok since this flow shouldn't be called often or in conjunction with other flows.